src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -115,8 +246,79 @@ public Uri GetEndpointForClient(ConfigurationClient client)
             }
 
             ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
+
+            if (currentClient == null)
+            {
+                currentClient = _dynamicClients.FirstOrDefault(c => c.Client == client);
+            }
             
             return currentClient?.Endpoint;
-        }
+        }
+
+        private async Task RefreshFallbackClients(CancellationToken cancellationToken)
+        {
+            IEnumerable<SrvRecord> results = Enumerable.Empty<SrvRecord>();
+
+            try
+            {
+                results = await _srvLookupClient.Value.QueryAsync(_endpoint.DnsSafeHost, cancellationToken).ConfigureAwait(false);
+            }
+            // Catch and log all exceptions thrown by srv lookup client to avoid new possible exceptions on app startup.
+            catch (SocketException ex)
+            {
+                _logger.LogWarning(LogHelper.BuildFallbackClientLookupFailMessage(ex.Message));
+                return;
+            }
+            catch (DnsResponseException ex)
+            {
+                _logger.LogWarning(LogHelper.BuildFallbackClientLookupFailMessage(ex.Message));
+                return;
+            }
+            catch (InvalidOperationException ex)
+            {
+                _logger.LogWarning(LogHelper.BuildFallbackClientLookupFailMessage(ex.Message));
+                return;
+            }
+            catch (DnsXidMismatchException ex)
+            {
+                _logger.LogWarning(LogHelper.BuildFallbackClientLookupFailMessage(ex.Message));

```

Comment:
Looks like _logger can be null.

Reply:
Perhaps validate for non-null when logger is set.

Reply:
Ensured won't set a null logger